### PR TITLE
Add example of tuning RocksDB on NNI

### PR DESCRIPTION
**Pull request by gxiaotian**
This PR adds example of tuning RocksDB on NNI as well as its English and Chinese documents.

### CODE REVIEW

File: docs/en_US/TrialExample/RocksdbExamples.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+`code directory: example/trials/systems/rocksdb-fillrandom`
+
+## Goals
+
+This example illustrates how to use NNI to search the best configuration of RocksDB for a `fillrandom` benchmark supported by a benchmark tool `db_bench`, which is a official benchmark tool provided by RocksDB itself. Therefore, before running this example, please make sure NNI is installed and [`db_bench`](https://github.com/facebook/rocksdb/wiki/Benchmarking-tools) is in your `PATH`. Please refer to [here](../Tutorial/QuickStart.md) for detailed information about installation and preparing of NNI environment, and [here](https://github.com/facebook/rocksdb/blob/master/INSTALL.md) for compiling RocksDB as well as `db_bench`.
```
Comment (xuehui1991) [inline docs/en_US/TrialExample/RocksdbExamples.md]: About the environment, shall we explain more detail or offer a script to run it?(suggestion)
Which means users could run one command to start this example.

File: docs/en_US/TrialExample/RocksdbExamples.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## Search Space
+
+For simplicity, this example tunes three parameters, `buffer_size`, `min_write_buffer_num` and `level0_file_num_compaction_trigger`, for writing 16M keys with 20 Bytes of key size and 100 Bytes of value size randomly, based on writing operations per second (OPS). The search space is specified by a `search_space.json` file as shown below. Detailed explanation of search space could be found [here](https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/SearchSpaceSpec.md).
+
+```json
+{
+    "write_buffer_size": {
```
Comment (xuehui1991) [inline docs/en_US/TrialExample/RocksdbExamples.md]: how these parameters affect the result?(suggestion)

File: examples/trials/systems/rocksdb-fillrandom/search_space.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+    "write_buffer_size": {
+        "_type": "quniform",
+        "_value": [2097152, 16777216, 1048576]
```
[the remark this answers is not in the document]
Comment (gxiaotian) [reply examples/trials/systems/rocksdb-fillrandom/search_space.json]: `db_bench` receives `write_buffer_size` in Bytes.

2097152 B = 2 MB
16777216 B = 16 MB
1048576 B = 1 MB

File: docs/en_US/TrialExample/RocksdbExamples.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+Here is the example of tuning RocksDB with TPE algorithm:
+
+`code directory: examples/trials/systems/rocksdb-fillrandom/config_tpe.yml`
+
+## Launch the experiment
+
+In order to run this example, you could enter the example folder and start the experiment using following commands:
+
+```bash
```
[the remark this answers is not in the document]
Comment (gxiaotian) [reply docs/en_US/TrialExample/RocksdbExamples.md]: Add soon~

File: docs/en_US/TrialExample/RocksdbExamples.md (hunk 1 above)
Comment (gxiaotian) [reply docs/en_US/TrialExample/RocksdbExamples.md]: The last section illustrated how to start this example. Shall we move it here?

File: docs/en_US/TrialExample/RocksdbExamples.md (hunk 2 above)
Comment (gxiaotian) [reply docs/en_US/TrialExample/RocksdbExamples.md]: Add soon~

**Comment by QuanluZhang** [comment]
looks great! the first example for tuning system parameters

File: docs/en_US/TrialExample/RocksdbExamples.md (hunk 1 above)
Comment (xuehui1991) [reply docs/en_US/TrialExample/RocksdbExamples.md]: I mean the part that how to build the environments of RocksDB.

File: examples/trials/systems/rocksdb-fillrandom/search_space.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (xuehui1991) [reply examples/trials/systems/rocksdb-fillrandom/search_space.json]: Got it.

File: docs/en_US/TrialExample/RocksdbExamples.md (hunk 1 above)
Comment (gxiaotian) [reply docs/en_US/TrialExample/RocksdbExamples.md]: I see~ Will add soon~

File: docs/en_US/TrialExample/RocksdbExamples.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (gxiaotian) [reply docs/en_US/TrialExample/RocksdbExamples.md]: Updated

File: docs/en_US/TrialExample/RocksdbExamples.md (hunk 1 above)
Comment (gxiaotian) [reply docs/en_US/TrialExample/RocksdbExamples.md]: Updated

File: docs/en_US/TrialExample/RocksdbExamples.md (hunk 2 above)
Comment (gxiaotian) [reply docs/en_US/TrialExample/RocksdbExamples.md]: Updated

**Comment by xuehui1991** [comment]
Hello @gxiaotian  seems there is some syntax error here:
"
./examples/trials/systems/rocksdb-fillrandom/main.py:29:24: E999 SyntaxError: invalid syntax
    args = [f"--{k}={v}" for k, v in parameters.items()]
                       ^
1     E999 SyntaxError: invalid syntax
"

**Comment by gxiaotian** [comment]
Hi @xuehui1991 , the script runs well in my experiments. I don't know why flake8 treats it a syntax error. Do you have any clue? Thank you!

**Comment by xuehui1991** [comment]
> Hi @xuehui1991 , the script runs well in my experiments. I don't know why flake8 treats it a syntax error. Do you have any clue? Thank you!

Could you install flake8 in your environment and reproduce this error?

**Comment by gxiaotian** [comment]
> > Hi @xuehui1991 , the script runs well in my experiments. I don't know why flake8 treats it a syntax error. Do you have any clue? Thank you!
> 
> Hello @gxiaotian Could you install flake8 in your environment and reproduce this error?

Hi @xuehui1991 , great thanks for your kind help. flake8 is my major linter for python, but I can't reproduce this error. Here are the details about my flake8: `3.7.8 (mccabe: 0.6.1, pycodestyle: 2.5.0, pyflakes: 2.1.1) CPython 3.6.8 on Linux` . Follows are the errors flake8 reminds me. All of them are `E501 line too long` in the MIT License part.

main.py:6:80: E501 line too long (110 > 79 characters)
main.py:7:80: E501 line too long (113 > 79 characters)
main.py:8:80: E501 line too long (114 > 79 characters)
main.py:9:80: E501 line too long (100 > 79 characters)
main.py:10:80: E501 line too long (128 > 79 characters)
main.py:12:80: E501 line too long (95 > 79 characters)
main.py:13:80: E501 line too long (92 > 79 characters)
main.py:14:80: E501 line too long (94 > 79 characters)
main.py:15:80: E501 line too long (96 > 79 characters)
main.py:16:80: E501 line too long (89 > 79 characters)

**Comment by xuehui1991** [comment]
> > > Hi @xuehui1991 , the script runs well in my experiments. I don't know why flake8 treats it a syntax error. Do you have any clue? Thank you!
> > 
> > 
> > Hello @gxiaotian Could you install flake8 in your environment and reproduce this error?
> 
> Hi @xuehui1991 , great thanks for your kind help. flake8 is my major linter for python, but I can't reproduce this error. Here are the details about my flake8: `3.7.8 (mccabe: 0.6.1, pycodestyle: 2.5.0, pyflakes: 2.1.1) CPython 3.6.8 on Linux` , and follows are the check results.
> 
> image

Hello @chicm-ms, could you help to check the flake version of the pipeline? Thank you~

**Pull request merged**